File: spring_security/eg05_sb3_oauth2_resource_server_jwt_symmetric_key/src/main/java/com/sc/sb/auth/entity/ScUserRole.java
package com.sc.sb.auth.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
@Table(name = "SC_USER_ROLE")
public class ScUserRole {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "ID")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "SC_USER_ID")
    private ScUser scUser;

    @ManyToOne
    @JoinColumn(name = "SC_ROLE_ID")
    private ScRole scRole;
}

File: spring_security/eg06_kotlin_sb3_oauth2_resource_server_jwt_symmetric_key/src/main/kotlin/com/sc/sb/auth/entity/ScUserRole.kt
package com.sc.sb.auth.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table

@Entity
@Table(name = "SC_USER_ROLE")
data class ScUserRole(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        @Column(name = "ID")
        val id: Long?,

        @ManyToOne
        @JoinColumn(name = "SC_USER_ID")
        val scUser: ScUser?,

        @ManyToOne
        @JoinColumn(name = "SC_ROLE_ID")
        val scRole: ScRole?

)
